src/Raven.Server/Documents/Indexes/Static/V8/JavaScriptMapOperation.cs

```diff
@@ -0,0 +1,143 @@
+            _engine = (V8EngineEx)_javaScriptIndexUtils.EngineHandle;
+
+            InternalHandle mapFuncV8Aux = mapFunc;
+            MapFunc = new InternalHandle(ref mapFuncV8Aux, true);
+            MapString = mapString;
+        }
+
+        ~JavaScriptMapOperation()
+        {
+            MapFunc.Dispose();
+        }
+
+        public override IEnumerable<InternalHandle> IndexingFunction(IEnumerable<object> items)
+        {
+            foreach (var item in items)
+            {
+                _engine.ResetCallStack();
+                _engine.ResetConstraints();
+
+                if (_javaScriptIndexUtils.GetValue(item, out JsHandle jsItemHandle) == false)
+                {
+                    continue;
+                }
+
+#if DEBUG
+                _engine.MakeSnapshot("map");
+#endif
+
+                var jsItem = jsItemHandle.V8.Item;
+                if (jsItem.IsBinder)
+                {
+                    //jsItem.SetReadyToDisposal();
+                    using (jsItem)
+                    {
+                        InternalHandle jsRes = InternalHandle.Empty;
+                        try
+                        {
+                            if (!MapFunc.IsFunction)
+                            {
+                                throw new JavaScriptIndexFuncException($"MapFunc is not a function");
+                            }
+                            jsRes = MapFunc.StaticCall(jsItem);
+                            jsRes.ThrowOnError();
+                        }
+                        catch (V8Exception jse)
+                        {
+                            var (message, success) = JavaScriptIndexFuncException.PrepareErrorMessageForJavaScriptIndexFuncException(MapString, jse);
+                            jsRes.Dispose();
+                            if (success == false)
+                                throw new JavaScriptIndexFuncException($"Failed to execute {MapString}", jse);
+                            throw new JavaScriptIndexFuncException($"Failed to execute map script, {message}", jse);
+                        }
+                        catch (Exception e)
+                        {
+                            jsRes.Dispose();
+                            throw new JavaScriptIndexFuncException($"Failed to execute {MapString}", e);
+                        }
+                        finally
+                        {
+                            _engine.ForceV8GarbageCollection();
+                        }
+
+                        using (jsRes)
+                        {
+                            if (jsRes.IsArray)
+                            {
+                                var length = (uint)jsRes.ArrayLength;
+                                for (int i = 0; i < length; i++)
+                                {
+                                    var arrItem = jsRes.GetProperty(i);
+                                    using (arrItem) { 
```

Comment:
Formatting, `{` on its own line

Reply:
done
Jint and V8 versions have been unified to single class based on JsHandle